src/main/java/duke/task/DeadlineTask.java

```diff
@@ -0,0 +1,43 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import duke.enums.TaskType;
+
+/**
+ * Represents a DeadlineTask class that deals with the deadline task that has a deadline `by` attached to it.
+ */
+public class DeadlineTask extends Task {
+    private final LocalDate by;
+
+    /**
+     * Constructor for DeadlineTask.
+     *
+     * @param description of the task.
+     * @param by          deadline of the task.
+     */
```

Comment:
I'm not sure if this follows the coding standard.
I would much prefer `@param description The description of the task`